### DISCUSSION TITLE
Error, laravel Docker CA Cert & Ghost CMS

### DISCUSSION THREAD

**Discussion by yukebrillianth** [planetscale/discussion#76] [Q&A] [answered]
I'm having trouble running migrations on Laravel, Ghost, and Strapi apps.

Because it uses foreign keys.

I'm learning to use PlanetScale DB, but I'm having problems:
1. Laravel Docker CA Cert
2. Strapi Foreign Key
3. Ghost Error Connection

**Comment by jgreet** [comment, marked as answer]
We've discussed foreign keys a few times here (https://github.com/planetscale/beta/discussions/18, https://github.com/planetscale/beta/discussions/74) and we're working on some more foreign key and foreign key constraint documentation to hopefully make that point even clearer.

As mentioned in those other posts, modeling your data with foreign keys will work just fine. Parent-child relationships, joining tables, all of that still works. It's foreign key CONSTRAINTs that are disallowed on PlanetScale, a trade-off we make to support no-downtime schema migrations.

How this changes your migrations in practice will depend on the framework. In Laravel for example, my understanding is that you can use the [foreignId](https://laravel.com/docs/8.x/migrations#column-method-foreignId) column type in your migrations but not the `constrained()` method, which adds the `CONSTRAINT... FOREIGN KEY` MySQL syntax that PlanetScale doesn't support.

**Comment by jgreet** [comment]
Oh, forgot to mention that we're also planning to add more framework-specific guides so it's good to hear what people are having trouble with. 

Anyone stuck getting their first Laravel migration running might find this video from @jamesmills helpful: https://youtu.be/1pO1a3IzWsA

**Reply by jamesmills** [reply]
Would love to get more feedback from people. Hope the video helps someone.

**Reply by yukebrillianth** [reply]
Thank you for making this video, it helped me a lot!

**Reply by yukebrillianth** [reply]
> Oh, forgot to mention that we're also planning to add more framework-specific guides so it's good to hear what people are having trouble with.
> 
> Anyone stuck getting their first Laravel migration running might find this video from @jamesmills helpful: https://youtu.be/1pO1a3IzWsA

have new problem on docker: PDO::__construct(): Unable to set private key file `/etc/ssl/certs/ca-certificates.crt'

**Reply by jgreet** [reply]
Are you sure that's the correct path for your Docker image? We list out a few of the common paths here:

https://docs.planetscale.com/reference/secure-connections#ca-root-configuration

It's also possible you'll need to install a `ca-certificates` package.

As a first step I'd open a shell in the Docker image as whatever account you're starting the app with and check that the file exists, and that the account has permission to read it.
